File: zig/src/inline/commentRule.zig
const std = @import("std");
const InlineParserState = @import("../types/InlineParserState.zig").InlineParserState;
const InlineRule = @import("../types/InlineRule.zig").InlineRule;
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const testCriticMarks = @import("criticMarksRule.zig").testCriticMarks;

pub fn testComment(state: *InlineParserState, parent: *MarkdownNode) bool {
    return testCriticMarks("comment", '>', state, parent, '<');
}

pub const commentRule = InlineRule{
    .name = "comment",
    .@"test" = testComment,
};

File: web/src/inline/commentRule.ts
import type InlineParserState from "../types/InlineParserState";
import type InlineRule from "../types/InlineRule";
import type MarkdownNode from "../types/MarkdownNode";
import testCriticMarks from "./criticMarksRule";

const rule: InlineRule = {
	name: "comment",
	test: testComment,
};
export default rule;

function testComment(state: InlineParserState, parent: MarkdownNode): boolean {
	return testCriticMarks(rule.name, ">", state, parent, "<");
}
